vod/mp4/mp4_parser.c

```diff
@@ -533,6 +545,54 @@ mp4_parser_parse_mdhd_atom(atom_info_t* atom_info, metadata_parse_context_t* con
 	return VOD_OK;
 }
 
+static vod_status_t
+mp4_parser_parse_udta_name_atom(atom_info_t* atom_info, metadata_parse_context_t* context)
+{
+	const char* name = (const char*)atom_info->ptr;
+	uint64_t name_len;
+
+	if (name == NULL) {
+		return VOD_OK;
+	}
+
+	// parse the name
+	if ((context->parse_params.parse_type & PARSE_FLAG_UDTA_NAME) == 0)
+	{
+		return VOD_OK;
+	}
+
+	// already set
+	if (context->media_info.label.data != NULL) {
+		return VOD_OK;
+	}
+
+	// assume '\0' terminated string
+	name_len = strlen(name);
```

Comment:
Must not run strlen on strings that are may not be null terminated. 
I think you can just set name_len to atom_info->size. But, if you have files where the atom data does contain trailing null terminators, then you can do something like this -
for (name_len = atom_info->size; name_len > 0 && name[name_len - 1] == '\0'; name_len--);

Reply:
In theory (probably considered a broken file) the atom may contain a null character elsewhere (not at the end), thats why I used `strlen()`. That null character will then end up in the manifest send to the client because the string is never checked again... Afaik other places parsing strings (like hdlr atom name) don't treat this case so I didnt either.